### ISSUE TITLE
Black doesn't support python 3.12.5, causing inline-snapshot to replace snapshot values with empty strings 

### ISSUE THREAD

**Issue by marklidenberg** [15r10nk/inline-snapshot#138]
# Code to Reproduce

On Python 3.12.5, the following code does nothing:

```python
from inline_snapshot import snapshot
import pytest


def test_something():
    assert 1 == snapshot() # will stay the same


if __name__ == "__main__":
    pytest.main(
        args=[
            __file__,
            f"--inline-snapshot=create",
        ]
    )
```

This code replaces a snapshot value with an empty string: 
```python
from inline_snapshot import snapshot
import pytest


def test_something():
    assert 1 == snapshot(2) # will update to assert 1 == snapshot()


if __name__ == "__main__":
    pytest.main(
        args=[
            __file__,
            f"--inline-snapshot=fix",
        ]
    )
```

---

# Cause

In `_format.py`:

```python
import warnings

from black import main
from click.testing import CliRunner


def format_code(text, filename):
    with warnings.catch_warnings():
        warnings.simplefilter("ignore")

        runner = CliRunner(mix_stderr=False)
        result = runner.invoke(
            main, ["--stdin-filename", str(filename), "-"], input=text
        )

    return result.stdout

```

`main` fails with the error:  
`Python 3.12.5 has a memory safety issue that can cause Black's AST safety checks to fail. Please upgrade to Python 3.12.6 or downgrade to Python 3.12.4.`

As a result, `result.stdout` is always an empty string.

**Comment by 15r10nk** [comment]
Thank you for this detailed report. I will handle the failure and write a warning which shows why the code could not be formatted.

**Comment by 15r10nk** [comment]
fixed in 0.14.1
